src/main/java/seedu/address/storage/JsonSerializableReminderList.java

```diff
@@ -0,0 +1,62 @@
+package seedu.address.storage;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.annotation.JsonRootName;
+
+import seedu.address.commons.exceptions.IllegalValueException;
+import seedu.address.model.reminder.ReadOnlyReminderList;
+import seedu.address.model.reminder.Reminder;
+import seedu.address.model.reminder.ReminderList;
+
+/**
+ * An Immutable ReminderList that is serializable to JSON format.
+ */
+@JsonRootName(value = "reminders")
+class JsonSerializableReminderList {
+
+    public static final String MESSAGE_DUPLICATE_REMINDER = "Reminder list contains duplicate reminder(s).";
+
+    private final List<JsonAdaptedReminder> reminders = new ArrayList<>();
+
+    /**
+     * Constructs a {@code JsonSerializableReminderList} with the given reminders.
+     */
+    @JsonCreator
+    public JsonSerializableReminderList(
+            @JsonProperty("reminders") List<JsonAdaptedReminder> reminders) {
+        this.reminders.addAll(reminders);
+    }
+
+    /**
+     * Converts a given {@code ReadOnlyReminderList} into this class for Jackson use.
+     *
+     * @param source future changes to this will not affect the created {@code JsonSerializableReminderList}.
+     */
+    public JsonSerializableReminderList(ReadOnlyReminderList source) {
+        reminders.addAll(source.getAllReminders().stream().map(JsonAdaptedReminder::new).collect(Collectors.toList()));
+    }
+
+    /**
+     * Converts this reminder list into the model's {@code ReminderList} object.
+     *
+     * @throws IllegalValueException if there were any data constraints violated.
+     */
+    public ReminderList toModelType() throws IllegalValueException {
+        ReminderList reminderList = new ReminderList();
+        for (JsonAdaptedReminder jsonAdaptedReminder : reminders) {
+            Reminder reminder = jsonAdaptedReminder.toModelType();
+            if (reminderList.contains(reminder)) {
+                throw new IllegalValueException(MESSAGE_DUPLICATE_REMINDER);
+            }
```

Comment:
why is it not allowed to have duplicated messages?

Reply:
i think it's because you cannot have 2 tasks with the same name, same time, etc.? it just doesn't make sense to have 2 meetings at the same time etc.

Reply:
I think I am enforcing it such that no reminders can be duplicated as @craeyeons mentioned.

Reply:
Would a set be preferred if no duplicates are allowed?

Reply:
> Would a observableSet be preferred if no duplicates are allowed? There would be problems with sorting though

Yes I think it would be preferred and there is a `SortedSet`, however, we will use the `SortedList` class to handle the sorting instead as it is more congruent with how we implemented the reminders in the UI.